api/src/main/java/uk/gov/dhsc/htbhf/claimant/entitlement/BackDatedCycleEntitlementCalculator.java

```diff
@@ -0,0 +1,74 @@
+import static java.util.Collections.min;
+
+/**
+ * Calculates the number of back dated vouchers a claimant is entitlement to.
+ * The expected due date and the birthdays of children resulting from the pregnancy are used to
+ * calculate the difference between the number of vouchers that were received for pregnancy and
+ * the number of vouchers that should have been received for the new children. If the difference in vouchers
+ * is less than zero, zero is returned.
+ */
+@Service
+public class BackDatedCycleEntitlementCalculator {
+
+    private final Integer entitlementCalculationDurationInDays;
+    private final EntitlementCalculator entitlementCalculator;
+
+    /**
+     * Constructor for {@link BackDatedCycleEntitlementCalculator}.
+     *
+     * @param paymentCycleConfig    configuration for the payment cycle
+     * @param entitlementCalculator calculates entitlement for a single calculation period
+     */
+    public BackDatedCycleEntitlementCalculator(PaymentCycleConfig paymentCycleConfig,
+                                               EntitlementCalculator entitlementCalculator) {
+        this.entitlementCalculationDurationInDays = paymentCycleConfig.getEntitlementCalculationDurationInDays();
+        this.entitlementCalculator = entitlementCalculator;
+    }
+
+    /**
+     * Calculates the back dated vouchers given the expected due date and the dates of births resulting from the pregnancy.
+     * @param expectedDueDate expected due date
+     * @param newChildrenDateOfBirths the dates of births of the children resulting from the pregnancy
+     * @return the number of back dated vouchers the claimant is entitled too
+     */
+    public Integer calculateBackDatedVouchers(Optional<LocalDate> expectedDueDate, List<LocalDate> newChildrenDateOfBirths) {
+        List<LocalDate> backDatedEntitlementDates = getBackDatedEntitlementDates(newChildrenDateOfBirths);
+        Integer vouchersForChildren = calculateNumberOfVouchers(Optional.empty(), newChildrenDateOfBirths, backDatedEntitlementDates);
+        Integer vouchersFromPregnancy = calculateNumberOfVouchers(expectedDueDate, emptyList(), backDatedEntitlementDates);
+
+        int backDatedVouchers = vouchersForChildren - vouchersFromPregnancy;
+        // do not return negative vouchers
+        return Math.max(backDatedVouchers, 0);
+    }
+
+    private int calculateNumberOfVouchers(Optional<LocalDate> expectedDueDate, List<LocalDate> newChildrenDateOfBirths, List<LocalDate> entitlementDates) {
+        return entitlementDates.stream()
+                .map(date -> entitlementCalculator.calculateVoucherEntitlement(expectedDueDate, newChildrenDateOfBirths, date))
+                .mapToInt(VoucherEntitlement::getTotalVoucherEntitlement)
+                .sum();
+    }
+
+    // get the list of entitlement dates since the oldest child was born
+    private List<LocalDate> getBackDatedEntitlementDates(List<LocalDate> newChildrenDatesOfBirth) {
```

Comment:
@mikeSmith08 Note that the logic in getBackDatedEntitlementDates has been changed, and tests updated because it was including one date _before_ the dob when it should only include dates >= dob